src/main/java/org/snomed/snowstorm/core/data/services/MultiSearchService.java

```diff
@@ -75,23 +63,39 @@ public class MultiSearchService implements CommitListener {
 	
 	private final Logger logger = LoggerFactory.getLogger(getClass());
 	
-	Map<String, String> publishedBranches = new HashMap<>();
-	
-	BoolQueryBuilder cachedBranchesQuery = null;
-	LocalDate cacheDate = null;
+	private final Map<String, String> publishedBranches = new HashMap<>();
 
-	public Page<Description> findDescriptions(DescriptionCriteria criteria, PageRequest pageRequest) {
-		
-		SearchHits<Description> searchHits = findDescriptionsHelper(criteria, pageRequest);
+	private MultiBranchCriteria cachedBranchCriteria = null;
+	private LocalDate cacheDate = null;
+
+	public Page<Description> findDescriptions(DescriptionCriteria criteria, String ecl, PageRequest pageRequest, Set<String> includeBranches) {
+
+		MultiBranchCriteria branchesQuery = new MultiBranchCriteria("all-released-plus-requested",getBranchesQuery().getTimepoint(),new ArrayList<>(getBranchesQuery().getBranchCriteria()));
```

Comment:
What's the importance of `all-released-plus-requested` @RichardAWood?

Reply:
It's just to distinguish it from the "all-released" MultiBranchCriteria used in other places (e.g. line 216).  It's relatively arbitrary, and could be set to something else if needed.

Reply:
I'm honestly not sure - it was there in the code when we started working on our changes.  Looking through the commit logs, it seems like it was added by Kai in late 2021 when getBranchesQuery() was changed from returning a BoolQueryBuilder to returning a MultiBranchCriteria.

SHA-1: 1d99e59c2deaae08ed56cf75f611be253a4f3685

* RT2-436 Add ecl parameter to description multisearch. Fix generics issues.

Reply:
Thanks @RichardAWood. I see what's happening now. 

The `all-released` is, like you said, an arbitrary branch path for grouping all versioned paths. `all-release-plus-requested` appends any requested paths onto this. 

I have noticed that if I include a versioned path in the request, then the versioned path will be queried twice. This does seem to stem from `MultiBranchCriteria::branchCriteria` being a `List` instead of a `Set` however.

Reply:
I don't have access to modify MultiBranchCriteria, so I've added a contains check to keep duplicate branches from being added:

		MultiBranchCriteria branchesQuery = new MultiBranchCriteria("all-released-plus-requested",cachedBranchesQuery.getTimepoint(),new ArrayList<>(cachedBranchesQuery.getBranchCriteria()));
		if (criteria.getIncludeBranches().getBranches() != null) {
			for(String branch : criteria.getIncludeBranches().getBranches()) {
				if (!branchesQuery.getBranchCriteria().contains(versionControlHelper.getBranchCriteria(branch))) {
					branchesQuery.getBranchCriteria().add(versionControlHelper.getBranchCriteria(branch));
				}
			}
		}